### PR TITLE
feat: Set is_highcpu flag based on platform

### PR DESCRIPTION
**Pull request by loewenheim**
This sets the `SaveEventTaskKind.is_highcpu` flag for an event based on whether the event's platform is included in `store.save-event-highcpu-platforms`.

Depends on https://github.com/getsentry/sentry/pull/54163.

### CODE REVIEW

File: src/sentry/tasks/store.py [hunk 1]
```diff
@@ -201,7 +201,9 @@ def _do_preprocess_event(
         return
 
     task_kind = SaveEventTaskKind(
-        has_attachments=has_attachments, from_reprocessing=from_reprocessing
+        has_attachments=has_attachments,
+        from_reprocessing=from_reprocessing,
+        is_highcpu=data["platform"] in options.get("store.save-event-highcpu-platforms"),
```
Comment (kamilogorek) [inline src/sentry/tasks/store.py:206]:
```suggestion
        is_highcpu=data["platform"] in options.get("store.save-event-highcpu-platforms", []),
```

File: src/sentry/tasks/store.py [hunk 2]
```diff
@@ -323,6 +325,7 @@ def _continue_to_save_event() -> None:
         task_kind = SaveEventTaskKind(
             from_reprocessing=process_task is process_event_from_reprocessing,
             has_attachments=has_attachments,
+            is_highcpu=data["platform"] in options.get("store.save-event-highcpu-platforms"),
```
Comment (kamilogorek) [inline src/sentry/tasks/store.py:328]:
```suggestion
            is_highcpu=data["platform"] in options.get("store.save-event-highcpu-platforms", []),
```

**Review by kamilogorek — CHANGES_REQUESTED** [review]
Better safe than sorry

**Comment by loewenheim** [comment]
> Better safe than sorry

Ain't that the truth.

**Review by Swatinem — APPROVED** [review]

**Review by kamilogorek — APPROVED** [review]

**Pull request merged**